rust/storage/src/admissioncontrolleds3.rs

```diff
@@ -143,10 +154,27 @@ impl AdmissionControlledS3Storage {
         }
     }
 
+    async fn enter(&self) -> SemaphorePermit<'_> {
+        match &*self.rate_limiter {
+            RateLimitPolicy::CountBasedPolicy(policy) => {
```

Comment:
I don't think we should leak knowledge of various policies into the Storage fwiw. Ideally this would be abstracted away and we impl RateLimitPolicy and present a unified view of policies. Maybe a TODO for when we introduce > 1 policy.

Reply:
Yeah I thought about this, I don't know what that unified view will look like once we have > 1 policy hence punted on it. I should add a TODO. TODOs are great at excusing "I'll check back later"

Reply:
Actually, I'll just move the enter() and exit() to impl with this interface now since it is small change and good to at least create the abstraction now